packaging/requirements/requirements-azure.txt

```diff
@@ -1,7 +1,7 @@
 packaging
 requests[security]
 azure-mgmt-compute>=2.0.0,<3
-azure-mgmt-network>=1.3.0,<2
+azure-mgmt-network>=1.6.0,<2
```

Comment:
I'm not seeing why we need to increase this value; the switch to all-kwargs seems to fix the problem even on `azure-mgmt-network==1.3.0`...

Reply:
My original intention is to make sure the Azure modules works well with latest SDK. In this case, it's 1.6.0. 

Actually, I don't know why previous contributors put a version range here. So, I just followed it. But, as we keep seeing more and more similar failures recently, I strongly suggest to use exact versions from now on, like what I did in PR #33897.


So, if you agree, I'd change this to 

```
azure-mgmt-network==1.6.0
```

Reply:
@nitzmahone

Reply:
The Azure requirements file was supposed to be a stopgap measure until the Azure SDK gets a better handle on the metapackage issues. I don't have strong feelings either way, but I'd prefer to let as many of them "float" as possible or specify only a minimum version, lest it turn into a new maintenance hassle. We're absolutely going to run into problems where modules need an API version that's not supported by a pinned version. 

I'm not sure what the best strategy is though; if we're going to pin specific versions, we should just pin *everything* and be done with it, since there seems to be an ongoing parade of breaking changes. That should hopefully solve the "stable code breaks every other week" issue, and just means we'd need to move the pins and deal with issues as we need new REST API versions.

@tstringer any opinions?

Reply:
I follow semantic versionning for release, and minor version should not break (or it's a bug). I know we got some issues with positional arguments (and we're moving to keyword-only arguments), but I really consider that as a bug.